src/append/rolling_file/policy/compound/trigger/time.rs

```diff
@@ -23,27 +22,27 @@ use crate::config::{Deserialize, Deserializers};
 #[derive(Copy, Clone, Eq, PartialEq, Hash, Debug, Default, serde::Deserialize)]
 #[serde(deny_unknown_fields)]
 pub struct TimeTriggerConfig {
+    /// The date/time interval between log file rolls.
     interval: TimeTriggerInterval,
+    /// Whether to modulate the interval.
     #[serde(default)]
     modulate: bool,
+    /// The maximum random delay in seconds.
     #[serde(default)]
     max_random_delay: u64,
 }
 
-#[cfg(not(feature = "config_parsing"))]
-/// Configuration for the time trigger.
-#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug, Default)]
-pub struct TimeTriggerConfig {
-    interval: TimeTriggerInterval,
-    modulate: bool,
-    max_random_delay: u64,
-}
-
```

Comment:
Please pull these fields back out of TimeTrigger.

Reply:
I did this because I want it to stay the same with `sizetrigger`:
`TimeTriggerConfig` is only used for deserialization. If `config_parse` is disabled, then we don't use this struct.
People who don't use configure file can use `TimeTrigger::new()` to build `builder` without extra struct.

It's ok to pull these back to TimeTriggerConfig. However, if I should change the logical of `SizeTriggerConfig` to be the same.

Reply:
@Dirreke its less about the actual change and more about having a clean PR. To me this PR would be best separated into:
1. rm deprecated chrono api call
2. change when trigger happens
3. refactor stuff

Whether each of those are net positive is TBD obviously but structuring this work that way would keep separate threads from being conflated.

Re this specific change: 
1. Separating static config from runtime state is important, thats why the config struct is useful.
2. Passing around those static values is error prone
3. This one I'm less sure of, I would guess that embedding a struct is more perf than destructuring and moving its content, but I'd love to see a godbolt link showing that or not.

The actual reason that this differs from sizetrigger is that there is a balance that we have to strike between managing complexity with abstraction and deconstruction of abstraction in pursuit of simplicity. In sizetrigger's case, there is no state and thus config is exactly equal to the runtime representation.

**TLDR; I stand by my original comment**